portal-impl/src/com/liferay/portlet/pluginsadmin/action/EditPluginAction.java

```diff
@@ -22,44 +24,47 @@
 import com.liferay.portal.security.auth.PrincipalException;
 import com.liferay.portal.service.PluginSettingServiceUtil;
 import com.liferay.portal.service.PortletServiceUtil;
-import com.liferay.portal.struts.PortletAction;
 import com.liferay.portal.util.PortalUtil;
+import com.liferay.portal.util.PortletKeys;
 
 import java.util.Arrays;
 
 import javax.portlet.ActionRequest;
 import javax.portlet.ActionResponse;
-import javax.portlet.PortletConfig;
-import javax.portlet.RenderRequest;
-import javax.portlet.RenderResponse;
-
-import org.apache.struts.action.ActionForm;
-import org.apache.struts.action.ActionForward;
-import org.apache.struts.action.ActionMapping;
 
 /**
  * @author Brian Wing Shun Chan
  * @author Jorge Ferrer
+ * @author Peter Fellwock
  */
-public class EditPluginAction extends PortletAction {
+
+@OSGiBeanProperties(
+	property = {
+		"javax.portlet.name=" + PortletKeys.PLUGINS_ADMIN,
+		"mvc.command.name=/plugins_admin/edit_plugin"
+	}
+)
+public class EditPluginAction extends BaseMVCActionCommand {
 
 	@Override
-	public void processAction(
-			ActionMapping actionMapping, ActionForm actionForm,
-			PortletConfig portletConfig, ActionRequest actionRequest,
-			ActionResponse actionResponse)
+	protected void doProcessAction(
+			ActionRequest actionRequest, ActionResponse actionResponse)
 		throws Exception {
 
 		try {
 			updatePluginSetting(actionRequest);
 
-			sendRedirect(actionRequest, actionResponse);
+			String redirect = ParamUtil.getString(actionRequest, "redirect");
+
+			sendRedirect(actionRequest, actionResponse, redirect);
 		}
 		catch (Exception e) {
 			if (e instanceof PrincipalException) {
 				SessionErrors.add(actionRequest, e.getClass());
 
-				setForward(actionRequest, "portlet.plugins_admin.error");
+				sendRedirect(
```

Comment:
I'm assuming that here you want to redirect to an error path. In that case you should do it with with the mvcPath renderParameter.

```java
actionResponse.setRenderParameter("mvcPath", "/html/portlet/plugins_admin/error.jsp");
```